ui/lib/sync/addon/routes/secrets/destinations/index.ts

```diff
@@ -5,22 +5,46 @@
 
 import Route from '@ember/routing/route';
 import { inject as service } from '@ember/service';
+import { hash } from 'rsvp';
 
 import type StoreService from 'vault/services/store';
+import type SyncDestinationModel from 'vault/vault/models/sync/destination';
 
 interface SyncSecretsDestinationsIndexRouteParams {
+  name: string;
+  type: string;
   page: string;
-  pageFilter: string;
 }
 
 export default class SyncSecretsDestinationsIndexRoute extends Route {
   @service declare readonly store: StoreService;
 
+  filterData(dataset: Array<SyncDestinationModel>, name: string, type: string): Array<SyncDestinationModel> {
+    let filteredDataset = dataset;
+    const filter = (key: keyof SyncDestinationModel, value: string) => {
+      return dataset.filter((model) => {
+        return model[key].toLowerCase().includes(value.toLowerCase());
+      });
+    };
+    if (name) {
+      filteredDataset = filter('name', name);
+    }
+    if (type) {
+      filteredDataset = filter('type', type);
+    }
+    return filteredDataset;
+  }
```

Comment:
Here's the custom filter method that is used in `lazyPaginatedQuery` so we can filter by type and name.